client/k8s/application/Application.go

```diff
@@ -220,25 +228,26 @@ func (impl K8sClientServiceImpl) GetPodLogs(restConfig *rest.Config, request *K8
 	return stream, nil
 }
 
-func (impl K8sClientServiceImpl) GetResourceIf(restConfig *rest.Config, request *K8sRequestBean) (resourceIf dynamic.NamespaceableResourceInterface, err error) {
+func (impl K8sClientServiceImpl) GetResourceIf(restConfig *rest.Config, request *K8sRequestBean) (resourceIf dynamic.NamespaceableResourceInterface, nonNamespacedKind []string, err error) {
 	resourceIdentifier := request.ResourceIdentifier
 	dynamicIf, err := dynamic.NewForConfig(restConfig)
 	if err != nil {
 		impl.logger.Errorw("error in getting dynamic interface for resource", "err", err)
-		return nil, err
+		return nil, nil, err
 	}
 	discoveryClient, err := discovery.NewDiscoveryClientForConfig(restConfig)
 	if err != nil {
 		impl.logger.Errorw("error in getting k8s client", "err", err)
-		return nil, err
+		return nil, nil, err
 	}
 	apiResource, err := ServerResourceForGroupVersionKind(discoveryClient, resourceIdentifier.GroupVersionKind)
```

Comment:
No need to call saparate API for this (getNonNamespacedResourceKind)

in apiResource, you get namespaced bool